gen/elem.go

```diff
@@ -605,3 +606,21 @@ func writeStructFields(s []StructField, name string) {
 		s[i].FieldElem.SetVarname(fmt.Sprintf("%s.%s", name, s[i].FieldName))
 	}
 }
+
+// coerceArraySize ensures we can compare constant array lengths.
+//
+// msgpack array headers are 32 bit unsigned, which is reflected in the
+// ArrayHeader implementation in this library using uint32. On the Go side, we
+// can declare array lengths as any constant integer width, which breaks when
+// attempting a direct comparison to an array header's uint32.
+//
+// If we have a string at this point, it represents a constant which could have
+// any integer width so we must coerce it to uint32 to guarantee it is
+// comparable.
+//
+func coerceArraySize(asz string) string {
+	if _, err := strconv.ParseInt(asz, 10, 64); err != nil {
```

Comment:
Typed constants can be base 16, too. Or base 8.

It is probably more robust to see if the constant is typed when parsing the AST.

Reply:
Ah cool, will have another dip tomorrow, see how it looks. Update this PR or submit a new one?

Reply:
If I can't find a neat way to do that during the AST parse, this could just check if the first character is a digit, or check if the token is a valid identifier and cover those cases too

Reply:
But that is definitely a bit more hacky than using the AST, if possible